lib/Doctrine/DBAL/Driver/AbstractSQLAnywhereDriver.php

```diff
@@ -46,6 +46,8 @@
     public function convertException($message, DriverException $exception)
     {
         switch ($exception->getErrorCode()) {
+            case '-306':
```

Comment:
I added those. The last one seems different as it disables the whole connection and not only the transaction.